Assets/MirageXR/Player/Scripts/Mobile/PopupsViewer.cs

```diff
@@ -92,10 +92,14 @@ private void OnOutTap()
 
     private void OnClose(PopupBase popup)
     {
-        if (_stack.Peek() == popup)
+        if (_stack.Count > 0)
```

Comment:
If you get an error from PopupViewer that the stack is empty, then you have done something wrong. 
which is how you use toggle when you change the toggle, the `ShowSmallCard `and `ShowBigCard `method will be called, which will cause the PopupViwer to close twice. This also causes ActivityListView to be updated twice, which takes a very long time. Ideally, we don't need to pull data from the server when we change the card display. We should just use the data we already have.

![image](https://user-images.githubusercontent.com/20320355/208152190-016d7363-7d8b-4c8d-b647-afe332557580.png)

But a check can be added.
```
    private void OnClose(PopupBase popup)
    {
        if (_stack.Count <= 0)
        {
            Debug.LogError("Stack is empty!");
            return;
        }

        if (_stack.Peek() == popup)
        {
            Destroy(_stack.Pop().gameObject);
            UpdateView();
        }
    }
```

Reply:
Ah I see the issue, just pushed a fix. Though these toggles call UpdateView in ActivityListView_v2 which destroys the existing ActivityListItems and remakes them using the list of session containers we already have, so isn't pulling data from the server every time we change the card size. Unless I am mistaken.